imagepipeline/src/main/java/com/facebook/imagepipeline/producers/DecodeProducer.java

```diff
@@ -122,6 +122,9 @@ public void run(EncodedImage encodedImage, boolean isLast) {
               ImageRequest request = producerContext.getImageRequest();
               if (mDownsampleEnabledForNetwork ||
                   !UriUtil.isNetworkUri(request.getSourceUri())) {
+                if (!EncodedImage.isMetaDataAvailable(encodedImage)) {
```

Comment:
@kirwan 
AddImageTransformMetaDataProducer is too late possibly. 
DecodeProducer's encodedImage.setSampleSize may be invalid.
And this problem  is not always occur.